skopt/utils.py

```diff
@@ -315,6 +315,7 @@ def cook_estimator(base_estimator, space=None, **kwargs):
     if base_estimator == "GP":
         if space is not None:
             space = Space(space)
+            space = Space(normalize_dimensions(space.dimensions))
```

Comment:
Having to do this is pretty ugly. The reason for doing this is that `normalize_dimensions` contains the logic for noticing that all dimensions are categorical and explicitly sets the transform to "identity". We should refactor this :-/